u8timeseries/models/RNN_model.py

```diff
@@ -0,0 +1,469 @@
+
+        if model in ['RNN', 'LSTM', 'GRU']:
+            hidden_fc_size = [] if hidden_fc_size is None else hidden_fc_size
+            self.model = RNNModule(name=model, input_size=self.input_size, hidden_dim=hidden_size,
+                                   num_layers=n_rnn_layers, output_length=output_length,
+                                   num_layers_out_fc=hidden_fc_size, dropout=dropout)
+        elif isinstance(model, nn.Module):
+            self.model = model
+        else:
+            raise ValueError('{} is not a valid RNN model.\n Please specify' \
+                             ' "RNN", "LSTM", "GRU", or give your own PyTorch nn.Module'.
+                             format(model.__class__.__name__))
+        self.model = self.model.to(self.device)
+        self.model_name = model_name
+        self.work_dir = work_dir
+
+        self.n_epochs = n_epochs
+        self.batch_size = batch_size
+        self.from_scratch = True  # do we train the model from scratch
+
+        # Define the loss function
+        self.criterion = loss_fn
+
+        # The tensorboard writer
+        self.tb_writer = None
+
+        # A utility function to create optimizer and lr scheduler from desired classes
+        def _create_from_cls_and_kwargs(cls, kws):
+            try:
+                instance = cls(**kws)
+            except (TypeError, ValueError) as e:
+                raise ValueError('Error when building the optimizer or learning rate scheduler;'
+                                 'please check the provided class and arguments'
+                                 '\nclass: {}'
+                                 '\narguments (kwargs): {}'
+                                 '\nerror:\n{}'.format(cls, kws, e))
+            return instance
+
+        # Create the optimizer and (optionally) the learning rate scheduler
+        if optimizer_kwargs is None:
+            optimizer_kwargs = {'lr': 1e-3}  # we do not set this as a default argument to avoid mutable values
+        optimizer_kwargs['params'] = self.model.parameters()
+        self.optimizer = _create_from_cls_and_kwargs(optimizer_cls, optimizer_kwargs)
+
+        if lr_scheduler_cls is not None:
+            lr_scheduler_kwargs['optimizer'] = self.optimizer
+            self.lr_scheduler = _create_from_cls_and_kwargs(lr_scheduler_cls, lr_scheduler_kwargs)
+        else:
+            self.lr_scheduler = None  # We won't use a LR scheduler
+
+    def fit(self,
+            series: TimeSeries,
+            val_series: Optional[TimeSeries] = None,
+            verbose: bool = False) -> None:
+        """
+        :param series: The training time series
+        :param val_series: Optionally, a validation time series that will
+                           be used to compute validation loss throughout training
+        """
+
+        super().fit(series)
+
+        if self.from_scratch:
+            shutil.rmtree(_get_checkpoint_folder(self.work_dir, self.model_name), ignore_errors=True)
```

Comment:
can be dangerous if variables are not set correctly.

Reply:
True... I don't like it too much. what would be a better way to manage our checkpoints folders?